measureme/src/counters.rs

```diff
@@ -543,16 +543,20 @@ mod hw {
                 asm!(
                     // Dummy `cpuid(0)` to serialize instruction execution.
                     "xor eax, eax",
+                    // LLVM sometimes reserves `ebx` for its internal use, we so we need to use
+                    // a scratch register for it instead.
+                    "mov {tmp_rbx:r}, rbx",
                     "cpuid",
+                    "mov rbx, {tmp_rbx:r}",
```

Comment:
I've noticed this. This is frustrating, as it makes `cpuid` undesirably troublesome (I wish Intel and AMD would agree on e.g. `mfence`/`lfence` being serializing). I suspect that using a LLVM intrinsic for `cpuid` would result in shorter codegen, can you try this? Either way, can you PM me on Zulip?

I've let that `rustc` PR sit, sadly (long story short, other factors forced us to abandon the grant it was part of), and it would be good to take stock of what's useful and what's just hacks we only needed to get the results we wanted.